fletcher/string_array.py

```diff
@@ -48,6 +49,39 @@ def __init__(self, array):
                 "Unsupported type passed for StringArray: {}".format(type(array))
             )
 
+    def argsort(self, ascending=True, kind="quicksort", *args, **kwargs):
+        """
+        Return the indices that would sort this array.
+
+        Parameters
+        ----------
+        ascending : bool, default True
+            Whether the indices should result in an ascending
+            or descending sort.
+        kind : {'quicksort', 'mergesort', 'heapsort', 'insertionsort'}, optional
+            Sorting algorithm.
+        *args, **kwargs:
+            passed through to :func:`numpy.argsort`.
+
+        Returns
+        -------
+        index_array : ndarray
+            Array of indices that sort ``self``.
+
+        See Also
+        --------
+        numpy.argsort : Sorting implementation used internally.
+        """
+        ascending = nv.validate_argsort_with_ascending(ascending, args, kwargs)
+        if kind not in ["quicksort", "insertionsort"]:
```

Comment:
this is not true, is it? Either this, or the exception message should tell something different